pkg/service/utgen/gen.go

```diff
@@ -116,9 +155,62 @@ func (g *UnitTestGenerator) Start(ctx context.Context) error {
 		} else {
 			g.cov.Current = 0
 		}
+
 		// Run the initial coverage to get the base line
 		iterationCount := 0
 		g.lang = GetCodeLanguage(g.srcPath)
+		if g.lang == "javascript" {
+			// Run 'npm test' and check if it works
+			g.logger.Info("Detected JavaScript. Running npm test...")
+
+			cmd := exec.CommandContext(ctx, "npm", "test")
+			_, err := cmd.CombinedOutput()
+			if err != nil {
+				g.logger.Error("Error running npm test")
+				return fmt.Errorf("npm test failed: %v", err)
+			}
+			g.logger.Info("npm test succeeded")
+		}
+
+		if g.lang == "python" {
+			// Run 'pytest'
```

Comment:
```
	if g.srcPath == "" {
		if err := g.runCoverage(); err != nil {
			return err
		}
		if len(g.Files) == 0 {
			return fmt.Errorf("couldn't identify the source files. Please mention source file and test file using flags")
		}
	}

```

Since we already have this why do we need to run this command ?

And as shown in the image we would now that test is already broken, why don't we use this instead of handling each case which is repititive
<img width="1150" alt="image" src="https://github.com/user-attachments/assets/773ec95f-ce76-4fd6-a545-26f5af7fe4cf">

Reply:
Oh okay got it

Reply:
this is still the same we should have used the pre-defined run coverage